Add OffScreenCanvas context lost/restored events (#33159)
mdn/content: files/en-us/web/api/offscreencanvas/contextlost_event/index.md

---
title: "OffscreenCanvas: contextlost event"
short-title: contextlost
slug: Web/API/OffscreenCanvas/contextlost_event
page-type: web-api-event
status:
  - experimental
browser-compat: api.OffscreenCanvas.contextlost_event
---

> {{APIRef}}{{SeeCompatTable}}

Reviewer: ```suggestion
{{APIRef("Canvas API")}}{{SeeCompatTable}}
```

Author: Thank you.

> The **`contextlost`** event of the [Canvas API](/en-US/docs/Web/API/Canvas_API) is fired if the user agent detects that the backing storage associated with a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.

Reviewer 2: Can we assume readers know what the "backing storage" means or does it matter here? Should we say "memory" instead?

> The **`contextlost`** event of the [Canvas API](/en-US/docs/Web/API/Canvas_API) is fired if the user agent detects that the backing storage associated with a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.

Reviewer: ```suggestion
The **`contextlost`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the user agent detects that the backing storage associated with a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.
```

normally behavior

> The **`contextlost`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the user agent detects that the backing storage associated with a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.

Author: ```suggestion
The **`contextlost`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser detects that the underlying resources used by the [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context for rendering are lost.
```

> The **`contextlost`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser detects that the underlying system that provides the resources used by a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.

Reviewer 2: ```suggestion
The **`contextlost`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser detects that the [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context is lost.
```

> Contexts can be lost for several reasons like driver crashes or the application runs out of memory, etc.

Reviewer 2: ```suggestion
Contexts can be lost for several reasons like process crashes, driver crashes, or the application runs out of memory, etc.
```

Author: ```suggestion
Contexts can be lost for several reasons, such as the GPU driver crashes, or the application runs out of memory, and so on.
```

By default the user agent will attempt to restore the context and then fire the [`contextrestored` event](/en-US/docs/Web/API/OffscreenCanvas/contextrestored_event).
User code can prevent the context from being restored by calling [`Event.preventDefault()`](/en-US/docs/Web/API/Event/preventDefault) during event handling.

## Syntax

Use the event name in methods like {{domxref("EventTarget.addEventListener", "addEventListener()")}}, or set an event handler property.

```js
addEventListener("contextlost", (event) => {});

oncontextlost = (event) => {};
```

## Event type

A generic {{domxref("Event")}}.

> ## Example

Reviewer: ```suggestion
## Examples
```

always *Examples*
